src/precompile.jl

```diff
@@ -107,6 +110,11 @@ function pkg_precompile()
             # end
         # end
     end
+    empty!(DEPOT_PATH)
+    empty!(LOAD_PATH)
+    append!(DEPOT_PATH, original_depot_path)
```

Comment:
Would it make sense to wrap the code into a `try ... finally ... end` block so that the paths are restored upon failure?

Reply:
If the precompilation process fails, there won't be any outputted precompile files, so does it matter at that point?

Reply:
I don't really see how that is relevant since the precompilation process runs in a separate process so any modifications to variables are only local for that process. So if this errors then we will immediately exit the process. 

In fact, I am not sure how this change does anything useful...